libsoftwarecontainer/src/gateway/dbus/dbusgateway.cpp

```diff
@@ -18,191 +17,100 @@
-
-    if (hasEnvVar) {
-        envVec.push_back(variable + "=" + envValue);
-    } else if (m_type == SessionProxy) {
-        log_error() << "Using DBus gateway in session mode"
-                    << " and no " + variable + " set in host environment, dbus-proxy won't work";
-        return false;
-    } else {
-        log_warn() << "Using DBus gateway in system mode"
-                   << " and no " + variable + " set in host environment, this could be a problem";
-    }
-
-    if (!startDBusProxy(commandVec, envVec)) {
-        return false;
-    }
-
-    // Write configuration
-    json_t *jsonConfig = json_object();
-    json_object_set(jsonConfig, DBusGatewayParser::SESSION_CONFIG, m_sessionBusConfig);
-    json_object_set(jsonConfig, DBusGatewayParser::SYSTEM_CONFIG, m_systemBusConfig);
-
-    char *config_c = json_dumps(jsonConfig, JSON_COMPACT);
-    std::string config = std::string(config_c);
-    json_decref(jsonConfig);
-    free(config_c);
-
-    return testDBusConnection(config);
+     return ReturnCode::SUCCESS;
 }
 
-bool DBusGateway::testDBusConnection(const std::string &config)
+ReturnCode DBusGateway::activate()
 {
-    log_debug() << "Sending config " << config;
-    unsigned int count = sizeof(char) * config.length();
-    ssize_t written = write(m_infp, config.c_str(), count);
-
-    // writing didn't work at all
-    if (written == -1) {
-        log_error() << "Failed to write to STDIN of dbus-proxy: " << strerror(errno);
-        return false;
-    } else if (written == (ssize_t)count) {
-        // writing has written exact amout of bytes
-        close(m_infp);
-        m_infp = INVALID_FD;
-        // dbus-proxy might take some time to create the bus socket
-        if (isSocketCreated()) {
-            log_debug() << "Found D-Bus socket: " << m_socket;
-            return true;
-        } else {
-            log_error() << "Did not find any D-Bus socket: " << m_socket;
-            return false;
-        }
-    } else {
-        // something went wrong during the write
-        log_error() << "Failed to write to STDIN of dbus-proxy!";
-        return false;
-    }
+     log_debug() << "Activating dbus session bus";
+     ReturnCode sessionBusActivationResult;
+     try {
+         sessionBusActivationResult = sessionBus.activate();
+     } catch (GatewayError e) {
+         sessionBusActivationResult = ReturnCode::FAILURE;
+     }
+
+     log_debug() << "Activating dbus system bus";
+     ReturnCode systemBusActivationResult ;
+     try {
+         sessionBusActivationResult = systemBus.activate();
+     } catch (GatewayError e) {
+         systemBusActivationResult = ReturnCode::FAILURE;
+     }
+
+     if (isError(sessionBusActivationResult) && isError(systemBusActivationResult)) {
+         std::string err = "Neither dbus session bus nor dbus system bus could be activated";
+         log_error() << err;
+         throw GatewayError(err);
+     }
+     return ReturnCode::SUCCESS;
```

Comment:
No since not being configured at activation will result in an exception being thrown.